cognite/client/data_classes/transformations/__init__.py

```diff
@@ -238,14 +309,68 @@ def try_get_or_create_nonce(oidc_credentials: Optional[OidcCredentials]) -> Opti
+                if source_oidc_credentials
+                else "DEFAULT"
+            )
+
+            ret = sessions_cache.get(key)
+            if not ret:
+                if source_oidc_credentials and source_oidc_credentials.client_id and source_oidc_credentials.client_secret:
+                    credentials = ClientCredentials(source_oidc_credentials.client_id, source_oidc_credentials.client_secret)
+                else:
+                    credentials = None
+                try:
+                    session = self._cognite_client.iam.sessions.create(credentials)
+                    ret = SourceNonceCredentials(session.id, session.nonce, self.source_oidc_credentials.cdf_project_name)
+                    sessions_cache[key] = ret
+                except Exception:
+                    ret = None
+            return ret
+
+        def try_get_or_create_destination_nonce(destination_oidc_credentials: Optional[DestinationOidcCredentials]) -> Optional[DestinationNonceCredentials]:
+            if keep_none and destination_oidc_credentials is None:
+                return None
+
+            # MyPy requires this to make sure it's not changed to None after inner declaration
+            assert sessions_cache is not None
+
+            key = (
+                f"{destination_oidc_credentials.client_id}:{hash(destination_oidc_credentials.client_secret)}"
+                if destination_oidc_credentials
+                else "DEFAULT"
+            )
+
+            ret = sessions_cache.get(key)
+            if not ret:
+                if destination_oidc_credentials and destination_oidc_credentials.client_id and destination_oidc_credentials.client_secret:
+                    credentials = ClientCredentials(destination_oidc_credentials.client_id, destination_oidc_credentials.client_secret)
+                else:
+                    credentials = None
+                try:
+                    session = self._cognite_client.iam.sessions.create(credentials)
+                    ret = DestinationNonceCredentials(session.id, session.nonce, self.destination_oidc_credentials.cdf_project_name)
+                    sessions_cache[key] = ret
+                except Exception:
+                    ret = None
+            return ret
+
+        if self.source_nonce:
+            self.source_nonce = try_get_or_create_source_nonce(self.source_oidc_credentials)
+            if self.source_nonce is None:
                 self.source_oidc_credentials = None
 
-        if self.destination_api_key is None and self.destination_nonce is None:
-            self.destination_nonce = try_get_or_create_nonce(self.destination_oidc_credentials)
-            if self.destination_nonce:
+        if  self.destination_nonce:
+            self.destination_nonce = try_get_or_create_destination_nonce(self.destination_oidc_credentials)
+            if self.destination_nonce is None:
                 self.destination_oidc_credentials = None
```

Comment:
These will also be redundant when removing SourceNonceCredentials and DestinantionNonceCredentials.

Reply:
Two classes `SourceNonceCredentials `and `SourceNonceCredentials `defined because, client OIDC config override read and write authentication because of using `NonceCredentials `class . That was the issue. I will describe the issue and the fix beefily in a new PR, since this PR includes data centric destination and class renaming deployments.